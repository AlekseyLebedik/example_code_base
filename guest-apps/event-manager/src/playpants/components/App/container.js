import React, { useEffect, lazy, Suspense } from 'react';
import isEqual from 'lodash/isEqual';
import { bindActionCreators } from 'redux';
import { useSelector } from 'react-redux';
import { connect } from 'dw/core/helpers/component';
import PropTypes from 'prop-types';
import { isEmpty } from 'lodash';

import CircularProgress from '@material-ui/core/CircularProgress';
import { useEMPermissions } from 'playpants/hooks';
import { ROUTES, BASE_PATH } from './components/Navigation/routes';
import { setProjectFromUnsafeSource } from './components/ProjectSelector/actions';
import {
  currentProjectSelector,
  configuredProjectsSelector,
  hasCurrentProjectSettingsSelector,
  isMissingSettingsSelector,
  isConfiguredSelector,
} from './selectors';
import * as Actions from './actions';

const StatelessApp = lazy(() => import('./presentational'));

export const App = ({
  fetchProjectSettings,
  fetchSchedules,
  fetchStories,
  fetchTemplates,
  match: {
    params: { projectId },
  },
  setProject,
}) => {
  const [loading, error, permissions] = useEMPermissions();
  const hasCurrentProjectSettings = useSelector(
    hasCurrentProjectSettingsSelector
  );
  const currentProject = useSelector(currentProjectSelector, isEqual);
  const configuredProject = useSelector(configuredProjectsSelector, isEqual);
  const isMissingSettings = useSelector(isMissingSettingsSelector);
  const isConfigured = useSelector(isConfiguredSelector, isEqual);
  /** Sets project, and loads users */
  useEffect(() => {
    if (!isEmpty(configuredProject)) {
      const pID = parseInt(projectId, 10);
      if (configuredProject.includes(pID)) {
        setProject(projectId);
      } else {
        setProject(configuredProject[0]);
      }
    }
  }, [projectId, configuredProject]);

  /** Fetches project settings when current project is changed or initialized */
  useEffect(() => {
    const { id: project } = currentProject;
    if (project && !isEmpty(configuredProject)) {
      fetchProjectSettings({ project__in: `${project},null` });
      fetchTemplates({ project });
      fetchStories({ project });
      fetchSchedules({ project });
    }
  }, [currentProject.id]);
  return !loading && !error ? (
    <Suspense fallback={<CircularProgress size={80} thickness={5} />}>
      <StatelessApp
        basePath={BASE_PATH}
        currentProject={currentProject}
        hasCurrentProjectSettings={hasCurrentProjectSettings}
        isMissingSettings={isMissingSettings}
        routes={ROUTES(permissions, hasCurrentProjectSettings, isConfigured)}
      />
    </Suspense>
  ) : null;
};

const mapDispatchToProps = dispatch => ({
  setProject: bindActionCreators(setProjectFromUnsafeSource, dispatch),
  fetchProjectSettings: bindActionCreators(
    Actions.fetchProjectSettings,
    dispatch
  ),
  fetchTemplates: bindActionCreators(Actions.fetchTemplates, dispatch),
  fetchStories: bindActionCreators(Actions.fetchStories, dispatch),
  fetchSchedules: bindActionCreators(Actions.fetchSchedules, dispatch),
});

App.propTypes = {
  fetchProjectSettings: PropTypes.func.isRequired,
  fetchSchedules: PropTypes.func.isRequired,
  fetchStories: PropTypes.func.isRequired,
  fetchTemplates: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired,
  setProject: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps, App);
