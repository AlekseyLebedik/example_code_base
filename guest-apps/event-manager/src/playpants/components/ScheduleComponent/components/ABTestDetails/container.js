import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import isEmpty from 'lodash/isEmpty';
import { withStyles } from '@material-ui/core/styles';

import ModalHandlers from 'dw/core/components/ModalHandlers';
import { timezoneOrDefaultSelector } from 'playpants/helpers/dateTime';

import { selectedViewSelector } from 'dw/core/components/EventsCalendar/selectors';
import eventSummaryStyles from 'playpants/components/EventSummaryDialog/styles';

import ABTestDetailsStateless from './presentational';
import * as display from './helpers';

export const ABTestDetails = props => {
  const { event, userTimezone } = props;

  if (isEmpty(event)) {
    return null;
  }

  const { creator, environment, id, testPeriodFrom, testPeriodTo, titleID } =
    event;

  const testLink = `/abtesting/view/${titleID}/${environment}/${id}`;
  const testPeriodToDate = display.displayViewTestDate(
    testPeriodTo,
    userTimezone
  );
  const testPeriodFromDate = display.displayViewTestDate(
    testPeriodFrom,
    userTimezone
  );
  const eventDuration = display.displayViewEventDuration(
    testPeriodTo,
    testPeriodFrom,
    userTimezone
  );

  const newProps = {
    ...props,
    testLink,
    testPeriodToDate,
    testPeriodFromDate,
    eventDuration,
    creator,
  };

  return <ABTestDetailsStateless {...newProps} />;
};

ABTestDetails.propTypes = {
  classes: PropTypes.object.isRequired,
  event: PropTypes.object.isRequired,
  openEventInNewTab: PropTypes.bool,
  userTimezone: PropTypes.string.isRequired,
  selectedView: PropTypes.string.isRequired,
};

ABTestDetails.defaultProps = {
  openEventInNewTab: false,
};

const mapStateToProps = (state, props) => ({
  userTimezone: timezoneOrDefaultSelector(state),
  selectedView: selectedViewSelector(state),
  visible: ModalHandlers.isVisibleSelector(state, props.baseModalId),
});

const mapDispatchToProps = (dispatch, props) => ({
  onOpen: () => dispatch(ModalHandlers.open(props.baseModalId)),
  onClose: () => dispatch(ModalHandlers.close(props.baseModalId)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(eventSummaryStyles)(ABTestDetails));
