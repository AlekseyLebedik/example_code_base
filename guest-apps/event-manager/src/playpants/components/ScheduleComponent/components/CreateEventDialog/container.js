import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { reset } from 'redux-form';

import { connect } from 'dw/core/helpers/component';
import { getNowTimestamp } from 'playpants/helpers/dateTime';
import ModalHandlers from 'dw/core/components/ModalHandlers';
import { tagFiltersHelpers } from 'dw/core/components/EventsCalendar/helpers';

import {
  currentProjectSelector,
  hasCurrentProjectSettingsSelector,
  storiesDataSelector,
  templatesDataSelector,
} from 'playpants/components/App/selectors';
import { prettyPrint } from 'playpants/helpers/json';

import {
  createEvent,
  setSelectedTemplate,
  resetCreateDialog,
} from 'playpants/components/ScheduleComponent/actions';
import { validateTag } from 'playpants/components/ScheduleComponent/helpers';
import {
  makeInitialValuesSelector,
  selectedTemplateSelector,
  templateSourceEventDataSelector,
  templateSourceEventLoadingSelector,
} from 'playpants/components/ScheduleComponent/selectors';

import CreateEventDialogStateless from './presentational';

export class CreateEventDialog extends Component {
  static propTypes = {
    baseUrl: PropTypes.string.isRequired,
    closeModal: PropTypes.func.isRequired,
    createEvent: PropTypes.func.isRequired,
    currentProject: PropTypes.object.isRequired,
    defaultToInfo: PropTypes.bool.isRequired,
    formName: PropTypes.string.isRequired,
    hasCurrentProjectSettings: PropTypes.bool.isRequired,
    history: PropTypes.object.isRequired,
    onResetCreateDialog: PropTypes.func.isRequired,
    onSetSelectedTemplate: PropTypes.func.isRequired,
    selectedTemplate: PropTypes.object.isRequired,
    templatesData: PropTypes.arrayOf(PropTypes.object).isRequired,
    templateSourceEventData: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.object,
    ]).isRequired,
    templateSourceEventLoading: PropTypes.bool.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {
      formData: {},
      globalChecked: !props.hasCurrentProjectSettings,
      globalLocked: !props.hasCurrentProjectSettings,
      repeatingEvent: false,
      scheduleWarningOpen: false,
      tags: [],
      tagsError: null,
      tagText: '',
      toggleAdvancedOn: false,
      toggleTypeOn: props.defaultToInfo,
    };
  }

  onResetAll = () => {
    this.props.onResetCreateDialog();
    reset(this.props.formName);
    this.clearTagEntry();
  };

  onCancel = () => {
    this.props.closeModal();
    this.onResetAll();
  };

  onChangeTag = tagText => {
    this.setState(state => ({
      tagText,
      tagsError: validateTag(state.tags, tagText),
    }));
  };

  setToggleAdvancedOn = () => {
    this.setState(state => ({
      toggleAdvancedOn: !state.toggleAdvancedOn,
    }));
  };

  setToggleTypeOn = () => {
    this.setState(state => ({
      toggleTypeOn: !state.toggleTypeOn,
      globalChecked: false,
    }));
  };

  setGlobalCheckbox = () => {
    this.setState(state => ({ globalChecked: !state.globalChecked }));
  };

  freeGlobalLocked = () => {
    if (this.props.hasCurrentProjectSettings) {
      this.setState({ globalLocked: false });
    }
  };

  requireGlobalCheckbox = () => {
    this.setState({ globalChecked: true, globalLocked: true });
  };

  initializeTags = tags => {
    this.setState({
      tags,
      tagText: '',
    });
  };

  clearTagEntry = () => {
    this.setState({
      tags: [],
      tagText: '',
    });
  };

  addTag = tagText => {
    this.setState(state => {
      const tagsError = validateTag(state.tags, tagText);
      if (tagsError) {
        return {
          tagsError,
        };
      }

      return {
        tags: [...state.tags, ...tagFiltersHelpers.extractTags(tagText)],
        tagText: '',
        tagsError: null,
      };
    });
  };

  deleteTag = tag => {
    this.setState(state => ({
      tags: state.tags.filter(t => t !== tag),
      tagText: '',
    }));
  };

  toggleScheduleWarningOpen = () => {
    this.setState(state => ({
      scheduleWarningOpen: !state.scheduleWarningOpen,
    }));
  };

  submitCreateEvent = (data = this.state.formData) => {
    this.props.createEvent(
      this.props.baseUrl,
      data,
      this.props.history,
      this.props.formName,
      this.onCancel
    );
  };

  handleFormSubmit = values => {
    // form might not have a selected template, so check first and continue
    const { currentProject, selectedTemplate, templateSourceEventData } =
      this.props;
    const storyValue = values.story ? { id: values.story } : values.story;
    const { tags, tagText, toggleTypeOn } = this.state;
    const templateActivities =
      templateSourceEventData && templateSourceEventData.activities;
    const isRestricted =
      selectedTemplate && selectedTemplate.restrict_activities;
    const activities = templateActivities
      ? templateActivities.map(item => ({
          type: item.type,
          name: item.name,
          activity: prettyPrint(item.activity),
          titles: item.titles,
          publish_on: item.publish_on,
          exec_order: item.exec_order,
        }))
      : [];
    const endDate = values.eventDates?.endDate || null;
    const tagsList = [...tags];
    if (tagText && !validateTag(tags, tagText)) {
      tagsList.push(tagText);
    }
    let project = currentProject.id;
    const projects = values.eventProjects;
    if (toggleTypeOn) {
      project = projects && projects.length === 1 ? projects[0] : null;
    }
    const repeatEventSettings =
      values.eventRecurrence === 'repeat' &&
      values.eventRepeatFrequency &&
      values.eventRepeatInterval &&
      values.eventRepeatEndDate
        ? {
            end_repeat_at: values.eventRepeatEndDate,
            frequency: Number(values.eventRepeatFrequency),
            interval: values.eventRepeatInterval,
            iteration: 0,
          }
        : null;

    const formData = {
      activities,
      authorizations: [],
      components: [],
      created_at: getNowTimestamp(),
      end_at: endDate,
      env_type: values.eventEnvType,
      note: values.eventNotes,
      project,
      publish_at: values.eventDates?.startDate,
      title: values.eventName,
      manual_tags: prettyPrint(tagsList),
      auto_tags: prettyPrint([]),
      platforms: prettyPrint(values.eventPlatforms || []),
      event_type: values.eventType || 'event-manager',
      is_restricted: !!isRestricted,
      repeat_event_settings: repeatEventSettings,
      story: storyValue,
      ...(values.isSchedule && { is_schedule: values.isSchedule }),
      ...(projects && { projects }),
    };
    if (formData.is_schedule && values.schedule) {
      this.setState({ formData });
      this.toggleScheduleWarningOpen();
    } else {
      this.submitCreateEvent(formData);
    }
  };

  render() {
    const newProps = {
      ...this.props,
      ...this.state,
      addTag: this.addTag,
      clearTagEntry: this.clearTagEntry,
      deleteTag: this.deleteTag,
      freeGlobalLocked: this.freeGlobalLocked,
      handleFormSubmit: this.handleFormSubmit,
      initializeTags: this.initializeTags,
      onCancel: this.onCancel,
      onChangeTag: this.onChangeTag,
      requireGlobalCheckbox: this.requireGlobalCheckbox,
      setGlobalCheckbox: this.setGlobalCheckbox,
      setToggleAdvancedOn: this.setToggleAdvancedOn,
      setToggleTypeOn: this.setToggleTypeOn,
      submitCreateEvent: this.submitCreateEvent,
      toggleScheduleWarningOpen: this.toggleScheduleWarningOpen,
    };
    return <CreateEventDialogStateless {...newProps} />;
  }
}

const makeMapStateToProps = () => {
  const initialValuesSelector = makeInitialValuesSelector();
  const mapStateToProps = (state, props) => ({
    currentProject: currentProjectSelector(state),
    hasCurrentProjectSettings: hasCurrentProjectSettingsSelector(state),
    initialValues: initialValuesSelector(state, props),
    isModalVisible: ModalHandlers.isVisibleSelector(state, props.formName),
    selectedTemplate: selectedTemplateSelector(state),
    templatesData: templatesDataSelector(state),
    templateSourceEventData: templateSourceEventDataSelector(state),
    templateSourceEventLoading: templateSourceEventLoadingSelector(state),
    storiesData: storiesDataSelector(state),
  });
  return mapStateToProps;
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  closeModal: () => dispatch(ModalHandlers.close(ownProps.formName)),
  createEvent: (baseUrl, data, callback, formName, onCancel) =>
    dispatch(createEvent(baseUrl, { data }, callback, formName, onCancel)),
  onResetCreateDialog: () => dispatch(resetCreateDialog()),
  onSetSelectedTemplate: selectedTemplate =>
    dispatch(setSelectedTemplate(selectedTemplate)),
});

export default connect(
  makeMapStateToProps,
  mapDispatchToProps,
  CreateEventDialog
);
