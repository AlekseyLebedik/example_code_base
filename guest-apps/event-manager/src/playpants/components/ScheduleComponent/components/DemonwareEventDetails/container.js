import React, { useCallback, useEffect, useMemo } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import isEmpty from 'lodash/isEmpty';
import isEqual from 'lodash/isEqual';
import { withStyles } from '@material-ui/core/styles';

import ModalHandlers from 'dw/core/components/ModalHandlers';
import { timezoneOrDefaultSelector } from 'playpants/helpers/dateTime';
import { eventFetchDetailsSelector } from 'playpants/components/ScheduleComponent/selectors';
import {
  useFetchEventDetails,
  useTitleNameSelector,
} from 'playpants/components/ScheduleComponent/hooks';

import { selectedViewSelector } from 'dw/core/components/EventsCalendar/selectors';
import eventSummaryStyles from 'playpants/components/EventSummaryDialog/styles';

import DemonwareDetailsStateless from './presentational';

export const DemonwareDetails = props => {
  const fetchEventDetails = useFetchEventDetails();
  const fetchedData = useSelector(eventFetchDetailsSelector, isEqual);
  const { event: item, baseModalId } = props;
  const titleNameSelector = useTitleNameSelector();
  const event = useMemo(
    () => ({
      ...item,
      ...fetchedData.data,
      titleName: titleNameSelector(item.title_id),
    }),
    [fetchedData, item, titleNameSelector]
  );
  const dispatch = useDispatch();
  const onOpen = useCallback(
    () => dispatch(ModalHandlers.open(baseModalId)),
    [dispatch, baseModalId]
  );
  const onClose = useCallback(
    () => dispatch(ModalHandlers.close(baseModalId)),
    [dispatch, baseModalId]
  );
  const selectedView = useSelector(selectedViewSelector);
  const userTimezone = useSelector(timezoneOrDefaultSelector);
  const visible = useSelector(state =>
    ModalHandlers.isVisibleSelector(state, baseModalId)
  );
  useEffect(() => fetchEventDetails(item), [item, fetchEventDetails]);
  if (isEmpty(event)) {
    return null;
  }
  return (
    <DemonwareDetailsStateless
      {...props}
      event={event}
      onClose={onClose}
      onOpen={onOpen}
      selectedView={selectedView}
      userTimezone={userTimezone}
      visible={visible}
      loading={fetchedData.loading}
    />
  );
};

DemonwareDetails.propTypes = {
  classes: PropTypes.object.isRequired,
  event: PropTypes.object.isRequired,
  openEventInNewTab: PropTypes.bool,
  userTimezone: PropTypes.string.isRequired,
  selectedView: PropTypes.string.isRequired,
  baseModalId: PropTypes.string.isRequired,
};

DemonwareDetails.defaultProps = {
  openEventInNewTab: false,
};

export default withStyles(eventSummaryStyles)(DemonwareDetails);
