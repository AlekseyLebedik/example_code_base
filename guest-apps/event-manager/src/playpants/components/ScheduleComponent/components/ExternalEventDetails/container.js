import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import isEmpty from 'lodash/isEmpty';
import { withStyles } from '@material-ui/core/styles';

import ModalHandlers from 'dw/core/components/ModalHandlers';
import { timezoneOrDefaultSelector } from 'playpants/helpers/dateTime';

import { selectedViewSelector } from 'dw/core/components/EventsCalendar/selectors';
import eventSummaryStyles from 'playpants/components/EventSummaryDialog/styles';

import ExternalDetailsStateless from './presentational';

export const ExternalDetails = props => {
  const { event } = props;
  if (isEmpty(event)) {
    return null;
  }
  return <ExternalDetailsStateless {...props} />;
};

ExternalDetails.propTypes = {
  classes: PropTypes.object.isRequired,
  event: PropTypes.object.isRequired,
  openEventInNewTab: PropTypes.bool,
  userTimezone: PropTypes.string.isRequired,
  selectedView: PropTypes.string.isRequired,
};

ExternalDetails.defaultProps = {
  openEventInNewTab: false,
};

const mapStateToProps = (state, props) => ({
  selectedView: selectedViewSelector(state),
  userTimezone: timezoneOrDefaultSelector(state),
  visible: ModalHandlers.isVisibleSelector(state, props.baseModalId),
});

const mapDispatchToProps = (dispatch, props) => ({
  onOpen: () => dispatch(ModalHandlers.open(props.baseModalId)),
  onClose: () => dispatch(ModalHandlers.close(props.baseModalId)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(eventSummaryStyles)(ExternalDetails));
