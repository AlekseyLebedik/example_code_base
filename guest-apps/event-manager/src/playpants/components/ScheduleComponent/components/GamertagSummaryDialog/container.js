import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import isEmpty from 'lodash/isEmpty';

import {
  calculateGamertagStartTime,
  checkFormatDateTime,
  getDurationFromSeconds,
  getDurationSecondsFromNow,
  getSecondsFromDuration,
  timezoneOrDefaultSelector,
} from 'playpants/helpers/dateTime';
import ModalHandlers from 'dw/core/components/ModalHandlers';
import { currentProjectIdSelector } from 'playpants/components/App/selectors';
import {
  fetchGroupTimewarpSettings,
  updateGroupTimewarpSettings,
} from './actions';

import StatelessGamertagSummaryDialog from './presentational';

const GamertagSummaryDialog = ({
  fetchTimewarpSettings,
  onClose,
  onFetchGamertagGroups,
  project,
  updateTimewarpSettings,
  userTimezone,
  timewarpSettings,
  ...props
}) => {
  const {
    color,
    date_time: dateTime,
    time_delta: timeDelta,
    type,
  } = timewarpSettings || {};
  const { selectedGamertagGroup } = props;
  const { id: groupId } = selectedGamertagGroup;
  const [editMode, setEditMode] = useState(false);
  const [changesMade, setChangesMade] = useState(false);
  const [displayColorPicker, setDisplayColorPicker] = useState(false);
  const [timewarpState, setTimewarpState] = useState({
    color: '#000000',
    duration: {},
    startTime: null,
    timeDelta: null,
    type: '',
  });
  const isOffset = timewarpState.type === 'offset';

  const setCurrentSettings = () => {
    setTimewarpState({
      ...timewarpState,
      color,
      duration: getDurationFromSeconds(timeDelta),
      startTime:
        dateTime || timeDelta
          ? calculateGamertagStartTime(userTimezone, dateTime, timeDelta)
          : null,
      timeDelta,
      type,
    });
  };

  useEffect(() => {
    if (groupId) fetchTimewarpSettings(groupId);
  }, [groupId]);
  useEffect(() => {
    setCurrentSettings();
  }, [color, dateTime, timeDelta, type]);

  const refetchGamertagGroups = () => {
    onFetchGamertagGroups({ type: 'gamertag-management', project });
  };

  const handleClose = (...args) => {
    onClose(...args);
  };
  const handleCancel = () => {
    setEditMode(false);
    setChangesMade(false);
    setCurrentSettings();
  };
  const handleSave = () => {
    setEditMode(false);
    if (changesMade) {
      updateTimewarpSettings(groupId, {
        ...timewarpSettings,
        color: timewarpState.color,
        type: timewarpState.type,
        ...(isOffset
          ? { date_time: null, time_delta: timewarpState.timeDelta }
          : {
              date_time: checkFormatDateTime(timewarpState.startTime),
              time_delta: null,
            }),
      });
      refetchGamertagGroups();
    }
  };
  const handleSetStartTime = timestamp => {
    if (isOffset) {
      const delta = timestamp ? getDurationSecondsFromNow(timestamp) : 0;
      setTimewarpState({
        ...timewarpState,
        startTime: calculateGamertagStartTime(userTimezone, timestamp, delta),
        duration: getDurationFromSeconds(delta),
        timeDelta: delta,
      });
    } else {
      setTimewarpState({
        ...timewarpState,
        startTime: calculateGamertagStartTime(userTimezone, timestamp),
      });
    }
    setChangesMade(true);
  };
  const handleSetGroupColor = col => {
    setTimewarpState({ ...timewarpState, color: col });
    setChangesMade(true);
  };
  const handleSetType = ({ target: { value } }) => {
    setTimewarpState({
      ...timewarpState,
      ...(isOffset ? { startTime: null } : { timeDelta: null }),
      type: value,
    });
    setChangesMade(true);
  };
  const handleSetDuration = e => {
    const { name, value } = e.target;
    const [, field] = name.split('.');
    const duration = Object.assign(timewarpState.duration);
    duration[field] = parseInt(value, 10);
    const delta = getSecondsFromDuration(timewarpState.duration);
    setTimewarpState({
      ...timewarpState,
      duration,
      timeDelta: delta,
      startTime: calculateGamertagStartTime(userTimezone, null, delta),
    });
    setChangesMade(true);
  };

  return (
    <StatelessGamertagSummaryDialog
      {...props}
      displayColorPicker={displayColorPicker}
      editMode={editMode}
      handleCancel={handleCancel}
      handleClose={handleClose}
      handleSave={handleSave}
      handleSetDuration={handleSetDuration}
      handleSetGroupColor={handleSetGroupColor}
      handleSetStartTime={handleSetStartTime}
      handleSetType={handleSetType}
      isOffset={isOffset}
      setDisplayColorPicker={setDisplayColorPicker}
      setEditMode={setEditMode}
      timewarpState={timewarpState}
      userTimezone={userTimezone}
    />
  );
};

GamertagSummaryDialog.propTypes = {
  baseModalId: PropTypes.string.isRequired,
  baseUrl: PropTypes.string.isRequired,
  fetchTimewarpSettings: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  onFetchGamertagGroups: PropTypes.func.isRequired,
  project: PropTypes.number.isRequired,
  selectedGamertagGroup: PropTypes.object.isRequired,
  timewarpSettings: PropTypes.object.isRequired,
  updateTimewarpSettings: PropTypes.func.isRequired,
  userTimezone: PropTypes.string.isRequired,
};

const timewarpSettingsSelector = state => {
  const { timewarpSettings } = state.Components.ScheduleComponent;
  return timewarpSettings.data && !isEmpty(timewarpSettings.data)
    ? timewarpSettings.data
    : {};
};

const mapStateToProps = (state, props) => ({
  userTimezone: timezoneOrDefaultSelector(state),
  visible: ModalHandlers.isVisibleSelector(state, props.baseModalId),
  timewarpSettings: timewarpSettingsSelector(state),
  project: currentProjectIdSelector(state),
});

const mapDispatchToProps = (dispatch, props) => ({
  onOpen: () => dispatch(ModalHandlers.open(props.baseModalId)),
  onClose: () => dispatch(ModalHandlers.close(props.baseModalId)),
  fetchTimewarpSettings: id => dispatch(fetchGroupTimewarpSettings(id)),
  updateTimewarpSettings: (id, params) =>
    dispatch(updateGroupTimewarpSettings(id, params)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GamertagSummaryDialog);
