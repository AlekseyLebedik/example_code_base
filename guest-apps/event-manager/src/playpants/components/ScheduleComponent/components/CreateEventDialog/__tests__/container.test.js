import React from 'react';
import { shallow } from 'enzyme';

import { createEventProps as props } from 'playpants/testUtils/scheduleProps';

import { CreateEventDialog } from '../container';

describe('CreateEvent', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow(<CreateEventDialog {...props} />);
    instance = wrapper.instance();
  });

  it('renders default values', () => {
    wrapper = shallow(<CreateEventDialog {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('clearTagEntry', () => {
    it('should clear tags and tagText', () => {
      wrapper.instance().clearTagEntry();
      expect(instance.state.tags).toEqual([]);
      expect(instance.state.tagText).toEqual('');
    });
  });

  describe('initializeTags', () => {
    it('should set tagText to given tag', () => {
      instance.initializeTags(['test', 'Call of Duty']);
      expect(instance.state.tags).toEqual(['test', 'Call of Duty']);
      expect(instance.state.tagText).toEqual('');
    });
  });

  describe('onChangeTag', () => {
    it('should set tagText to given tag', () => {
      instance.onChangeTag('test');
      expect(instance.state.tagText).toEqual('test');
    });
  });

  describe('addTag', () => {
    it("shouldn't add a tag to the eventTags array if given a blank string", () => {
      instance.addTag('');
      expect(instance.state.tags).toEqual([]);
      expect(instance.state.tags.length).toEqual(0);
    });

    it('should add a tag to the tags array', () => {
      instance.addTag('test');
      expect(instance.state.tags).toEqual(['test']);
      expect(instance.state.tags.length).toEqual(1);
    });
  });

  describe('deleteTag', () => {
    it('should delete a given tag from tags', () => {
      instance.setState({ tags: ['test'] });
      instance.deleteTag('test');
      expect(instance.state.tags).toEqual([]);
      expect(instance.state.tags.length).toEqual(0);
      expect(instance.state.tagText).toEqual('');
    });
  });

  describe('handleFormSubmit instance', () => {
    it('should call createEvent', () => {
      const values = {
        eventDates: { startDate: 1541191064 },
        eventNotes: '',
        eventName: 'Test',
      };

      instance.handleFormSubmit(values);
      expect(props.createEvent).toHaveBeenCalledTimes(1);
    });
  });
});
