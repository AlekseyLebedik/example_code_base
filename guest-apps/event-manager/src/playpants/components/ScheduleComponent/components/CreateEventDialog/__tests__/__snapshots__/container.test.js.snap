// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEvent renders default values 1`] = `
<CreateEventDialogStateless
  addTag={[Function]}
  baseUrl="event-manager/1/"
  clearTagEntry={[Function]}
  closeModal={[MockFunction]}
  createEvent={[MockFunction]}
  currentProject={
    Object {
      "contentTypeId": 19,
      "id": 1,
      "name": "GTR Project",
      "titles": Array [
        Object {
          "environments": Array [
            Object {
              "contentTypeId": 8,
              "id": 2,
              "shortType": "cert",
              "type": "Certification",
              "usesABTesting": true,
              "usesAE": true,
              "usesAsyncMMP": false,
              "usesLegacyStore": false,
              "usesMarketplace": true,
              "usesObjectStore": true,
            },
            Object {
              "contentTypeId": 8,
              "id": 1,
              "shortType": "dev",
              "type": "Development",
              "usesABTesting": true,
              "usesAE": true,
              "usesAsyncMMP": false,
              "usesLegacyStore": false,
              "usesMarketplace": true,
              "usesObjectStore": true,
            },
          ],
          "id": 1,
          "name": "GTR-PS3",
          "platform": "PS3",
        },
      ],
    }
  }
  defaultToInfo={false}
  deleteTag={[Function]}
  displayInfoEvents={true}
  formData={Object {}}
  formName=""
  freeGlobalLocked={[Function]}
  globalChecked={false}
  globalLocked={false}
  handleFormSubmit={[Function]}
  hasCurrentProjectSettings={true}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  initialValues={
    Object {
      "eventDates": Object {
        "endDate": null,
        "startDate": null,
      },
      "eventEnvType": "",
      "eventName": "",
      "eventNotes": "",
      "eventProjects": Array [
        1,
      ],
      "eventRecurrence": "never",
      "eventRepeatFrequency": 1,
      "eventRepeatInterval": "days",
      "eventType": "",
      "isSchedule": null,
      "schedule": null,
      "story": null,
    }
  }
  initializeTags={[Function]}
  isModalVisible={true}
  onCancel={[Function]}
  onChangeTag={[Function]}
  onResetCreateDialog={[MockFunction]}
  onSetSelectedTemplate={[MockFunction]}
  repeatingEvent={false}
  requireGlobalCheckbox={[Function]}
  scheduleWarningOpen={false}
  selectedStory={
    Object {
      "id": 1,
    }
  }
  selectedTemplate={Object {}}
  setGlobalCheckbox={[Function]}
  setToggleAdvancedOn={[Function]}
  setToggleTypeOn={[Function]}
  submitCreateEvent={[Function]}
  tagText=""
  tags={Array []}
  tagsError={null}
  templateSourceEventData={Object {}}
  templateSourceEventLoading={false}
  templatesData={Array []}
  title="Create"
  toggleAdvancedOn={false}
  toggleScheduleWarningOpen={[Function]}
  toggleTypeOn={false}
/>
`;
